jena-arq/src/main/java/org/apache/jena/riot/other/G.java

```diff
@@ -845,4 +850,110 @@ public static <X> X calcTxn(Graph graph, Supplier<X> action) {
+//            it.forEachRemaining(graph::add);
+//        }
+    }
+
+    /**
+     * Test whether a graph contains by "same term", regardless of whether the graph
+     * implements "same value" or "same term".
+     */
+    public static boolean containsBySameTerm(Graph graph, Triple triple) {
+        return containsBySameTerm(graph, triple.getSubject(), triple.getPredicate(), triple.getObject());
+    }
+
+    /**
+     * Test whether a graph contains by "same term", regardless of whether the graph
+     * implements "same value" or "same term".
+     */
+    public static boolean containsBySameTerm(Graph graph, Node s, Node p, Node o) {
+        // Do direct for efficiency.
+        if ( ! graph.contains(s,p,o) )
+            return false;
+        // May have matched by value.  Do a term test find to restrict to RDF terms.
+        ExtendedIterator<Triple> iter = graph.find(s, p, o);
+        // Unless generalized RDF, only need to test object.
+        Predicate<Triple> predicate = (dataTriple) -> sameTermMatch(s, p, o, dataTriple);
+        iter = iter.filterKeep(predicate);
+        try {
+            return iter.hasNext();
+        } finally { iter.close(); }
+
+//    // For reference - just object
+//    if ( !o.isConcrete() || o.isLiteral() ) {
+//        Predicate<Triple> predicate = (t) -> sameTermMatch(o, t.getObject()) ;
+//        iter = iter.filterKeep(predicate) ;
+//    }
+
+    }
+
+    /**
+     * Match a ground triple (even ANY and variables are considered ground terms in the
+     * data triple) with S/P/O which can be wildcards (ANY or null).
+     */
+    public static boolean sameTermMatch(Node matchSubj, Node matchPred, Node matchObj, Triple dataTriple) {
+        return sameTermMatch(matchSubj, dataTriple.getSubject()) &&
+               sameTermMatch(matchPred, dataTriple.getPredicate()) &&
+               sameTermMatch(matchObj,  dataTriple.getObject());
+    }
+
+    /**
+     * Match a ground RDF Term (ANY and variables are considered ground terms in the
+     * data term) with a node which can be a wildcard (ANY or null).
+     * Language tags compare case-insensitively.
+     */
+    public static boolean sameTermMatch(Node match, Node data) {
+        if ( false ) {
+            if ( match==null || Node.ANY.equals(match) )
+                return true;
+            // If lang tags are compared case-insensitively.
+            return match.equals(data);
+        }
+
+        // Allow for case-insensitive language tag comparison.
+        // Two literals.
+        if ( ! Util.isLangString(data) || ! Util.isLangString(match) )
+            // Not both strings with lang tags.
+            return (match==null) || (match == Node.ANY) || match.equals(data) ;
+
+        // Language tags compare case insensitively.
+        String lex1 = data.getLiteralLexicalForm();
+        String lex2 = data.getLiteralLexicalForm();
+        String lang1 = data.getLiteralLanguage();
+        String lang2 = data.getLiteralLanguage();
```

Comment:
Doing just a quick review as I don't know much about this part of the code (and it's late) so this comment could be something silly, :bow: 

@afs, won't `lex1` and `lex2` be identical? I haven't checked the code to see if `getLiteralLexicalForm` will return a new object, but maybe one was supposed to use `match` instead? Ditto for the `lang` vars?

Reply:
Good catch - 1 should be "match".